### ISSUE TITLE
Spaces after flags and category 

### ISSUE THREAD

**Issue by SpartanTek** [CTFd/CTFd#851]
I would like it if CTFd would automatically remove the blank spaces after all flags and categories in the entry process.  

**Environment**:
  - CTFd Version: 2.0.3
  - Operating System: Ubuntu 16.04
  - Web Browser and Version: Chrome

**Comment by ColdHeat** [comment]
I agree with some of this but I don't think stripping flags is the right idea. Could people have flags with whitespace at the end? Do we want to prevent or allow that? I definitely do not want to have to have special cases for people who want to have whitespace at the start/end of the flag.

**Comment by ColdHeat** [comment]
I don't really like the solutions here and I don't think this is really that big of an issue.

**Comment by ColdHeat** [comment]
An interesting point is that CTFd has actually always stripped flags when they are submitted by users. Meaning that a flag ending in a space is actually impossible.

**Comment by ColdHeat** [comment]
Regarding the above:

https://github.com/CTFd/CTFd/blob/61d2bddaed393ce5be5a1e911be09522bebaf523/CTFd/plugins/challenges/__init__.py#L122

attempt, solve, and fail methods all used strip(). So we should be okay to strip regular static and regex flags. Other flags I will leave as default.
